### PR TITLE
[QuickPi][SensorStart] Created new version of save sensors at start

### PR DESCRIPTION
**Pull request by NicolasWent**
Created second version of save of sensors at start.

In this version, the sensors are saved when a modification occurs while we are not running. Then when calling sensors.restore, if we have a saved version of this sensor then we put it.

The way I coded it assumes that two sensors can't have the same name even if they are of different type.

### CODE REVIEW

File: pemFioi/quickpi/blocklyQuickPi_lib.js [hunk 1]
```diff
@@ -7933,8 +7943,43 @@ var getContext = function (display, infos, curLevel) {
...
+                    callsInTimeSlot: sensor.callsInTimeSlot,
+                    lastTimeIncrease: sensor.lastTimeIncrease,
+                    removed: sensor.removed,
+                    quickStore: sensor.quickStore
+                };
+            }
+        }
+    }
+
+    function _copyArray(src) {
```
Comment (mblockelet) [inline pemFioi/quickpi/blocklyQuickPi_lib.js]: You can just use the slice function (`src.slice()`) to do this, no need for a function.

File: pemFioi/quickpi/blocklyQuickPi_lib.js [hunk 2]
```diff
@@ -2931,18 +2931,28 @@ var getContext = function (display, infos, curLevel) {
...
-            sensor.state = null;
-            sensor.screenDrawing = null;
-            sensor.lastDrawnTime = 0;
-            sensor.lastDrawnState = null;
-            sensor.callsInTimeSlot = 0;
-            sensor.lastTimeIncrease = 0;
-            sensor.removed = false;
-            sensor.quickStore = null;
-
+            if (context.sensorsSaved[sensor.name]) {
```
Comment (mblockelet) [inline pemFioi/quickpi/blocklyQuickPi_lib.js]: You should restore the saved state only when in "user mode" (`context.autoGrading = false;` if i'm not forgetting something).

**Review by mblockelet — CHANGES_REQUESTED** [review]
There are small modifications to make sure it doesn't break other tasks, but aside from that it works properly now and I haven't encountered any issues.

File: pemFioi/quickpi/blocklyQuickPi_lib.js (hunk 2 above)
Comment (NicolasWent) [reply pemFioi/quickpi/blocklyQuickPi_lib.js]: Thank you, I forgot about this one

**Review by mblockelet — APPROVED** [review]
Thanks, looks all good to me.

**Pull request merged**